product/views/product_views.py

```diff
@@ -0,0 +1,102 @@
+"""Product related views."""
+
+from django.conf import settings
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404, redirect, render
+from django.template import loader
+from rest_framework.views import View
+
+from product.forms import ProductForm
+from product.models import Product, ProductImages
+
+
+class ProductListView(LoginRequiredMixin, View):
+    """List products view."""
+
+    def get(self, request):
+        """Render product list template.."""
+        products = Product.objects.all()
+        template = loader.get_template('product/list-products.html')
+        context = {
+            'products': products,
+        }
+        return HttpResponse(template.render(context, request))
+
+
+product_list_view = ProductListView.as_view()
+
+
+class ProductDetailView(LoginRequiredMixin, View):
+    """Get product view."""
+
+    def get(self, request, id):
+        """Render product detail template.."""
+        product = get_object_or_404(Product, id=id)
+        template = loader.get_template('product/product-detail.html')
+        context = {
+            'product': product,
+            'has_images': ProductImages.objects.filter(product=product).exists(),
+        }
+        return HttpResponse(template.render(context, request))
+
+
+product_detail_view = ProductDetailView.as_view()
+
+
+class ProductDeleteView(LoginRequiredMixin, View):
+    """Delete product view."""
+
+    def get(self, request, id):
```

Comment:
use post .

Reply:
> use post .

@Ayub-Khan  all our `delete` views are `get` type views, will it be okay having this one alone as `post`?

Reply:
ah update all to post.